html/dom/reflection.js

```diff
@@ -521,23 +601,23 @@ ReflectionTests.typeMap = {
             {"test": 6}, NaN, +Infinity, -Infinity, "\0",
             {toString:function() {return 2;}, valueOf: null},
             {valueOf:function() {return 3;}, toString: null}],
-        "domExpected": [minInt - 1, minInt, -36, -1, 0, 1, maxInt,
-                        maxInt + 1, maxUnsigned, maxUnsigned + 1, null,
-                        // Leading whitespace tests
-                        7, null, 7, 7, null, null,
-                        7, 7, null, null, null, null,
-                        null, null, null, null, null, null,
-                        null, null, null, null, null, null,
-                        null,
-                        // End leading whitespace tests
-                        null, null, 1.5, 5, 100, 0.5, null, null,
-                        null, null, null, null, null,
-                        2, 3],
-        // I checked that ES ToString is well-defined for all of these (I
-        // think).  Yes, String(-0) == "0".
+        "domExpected": function (val) {
+            var parsed = ReflectionTests.parseFloat(String(val));
+            if (parsed === false) {
+                return null;
+            }
+            return parsed;
+        },
         "idlTests":       [ -10000000000,   -1,  -0,   0,   1,   10000000000],
-        "idlDomExpected": ["-10000000000", "-1", "0", "0", "1", "10000000000"],
-        "idlIdlExpected": [ -10000000000,   -1,   0,   0,   1,   10000000000]
+        "idlDomExpected": function (val) {
+            return Number(val).toFixed();
```

Comment:
This is actually wrong, it should be just `Number(val)`, which would be the default behavior when idlDomExpected is absent.
This wouldn't change the behavior for the test values, but could be wrong should new values be tested.

https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#best-representation-of-the-number-as-a-floating-point-number

Reply:
I amended the commits and force-pushed. Hope this is OK for you.